tests/nn/data_parallel/test_fsdp_optimizer_utils.py

```diff
@@ -89,6 +90,12 @@ def _test_consolidated_optimizer(self, config, rank, group, optim_fn=torch.optim
 
         if fsdp.rank > 0:
             return
```

Comment:
should we also test the non-0 ranks at some point?

Reply:
added `assert sd is None` before return